crypto/s2n_drbg.c

```diff
@@ -139,9 +142,15 @@ int s2n_drbg_instantiate(struct s2n_drbg *drbg, struct s2n_blob *personalization
     /* Seed / update the DRBG */
     GUARD(s2n_drbg_seed(drbg, &ps));
 
-    /* After initial seeding, pivot to RDRAND if available and not overridden */
-    if (drbg->entropy_generator == NULL && s2n_cpu_supports_rdrand()) {
-        drbg->entropy_generator = s2n_get_rdrand_data;
+    if (mode == S2N_AES_128_CTR_NO_DF_PR || mode == S2N_AES_256_CTR_NO_DF_PR) {
+        /* After initial seeding, pivot to RDRAND if available and not overridden */
+        if (drbg->entropy_generator == NULL && s2n_cpu_supports_rdrand()) {
+            drbg->entropy_generator = s2n_get_rdrand_data;
+        }
```

Comment:
Just noticed that there's a missing `else` here. What happens if `(mode == S2N_AES_128_CTR_NO_DF_PR || mode == S2N_AES_256_CTR_NO_DF_PR)` is true, but `drbg->entropy_generator == NULL && s2n_cpu_supports_rdrand()` is false. It looks like `drbg` wouldn't be initialized.

Reply:
Good catch, decided it makes sense to track the preferred entropy generation function and prediction resistance separately.